modules/apps/transfer/client/cli/tx.go

```
@@ -89,7 +90,21 @@ to the counterparty channel. Any timeout set to 0 is disabled.`),
 				}
 
 				if timeoutTimestamp != 0 {
-					timeoutTimestamp = consensusState.GetTimestamp() + timeoutTimestamp
+					// use local clock time as reference time if it is later than the
+					// consensus state timestamp of the counter party chain, otherwise
+					// still use consensus state timestamp as reference
+					now := time.Now().UnixNano()
+					consensusStateTimestamp := consensusState.GetTimestamp()
+					if now > 0 {
```

why would now be 0?

---

I was being very defensive. It could potentially be 0 if the clock of the machine is (for who knows what reason) on Jan 1st 1970 12:00 AM. And it could be negative if the clock is set before that date time.

---

Perhaps we can leave a comment explaining this? :)

---

I will do that, yes.

---

Maybe we should return an error if now is < 0?

---

Initially I also thought of returning an error, but then I realized that it was a situation where we could still do our best effort to fulfill the request by using the consensus state timestamp.

---

Yes, but this could cause problems in other areas of the codebase?

---

All right, I can change it. No problem.

---

@colin-axner I pushed this change. Do you approve it the way it is now? If yes, then I will merge.